payjoin/src/receive/mod.rs

```diff
@@ -559,27 +577,19 @@ impl ProvisionalProposal {
                 }
             }
         }
-
-        for i in sender_input_indexes {
-            log::trace!("Clearing sender input {}", i);
-            self.payjoin_psbt.inputs[i].non_witness_utxo = None;
-            self.payjoin_psbt.inputs[i].witness_utxo = None;
-            self.payjoin_psbt.inputs[i].final_script_sig = None;
-            self.payjoin_psbt.inputs[i].final_script_witness = None;
-            self.payjoin_psbt.inputs[i].tap_key_sig = None;
-        }
-        Ok(PayjoinProposal {
-            payjoin_psbt: self.payjoin_psbt,
-            owned_vouts: self.owned_vouts,
-            params: self.params,
-        })
+        return sender_input_indexes;
     }
 
     pub fn finalize_proposal(
         mut self,
         wallet_process_psbt: impl Fn(&Psbt) -> Result<Psbt, Error>,
         min_feerate_sat_per_vb: Option<FeeRate>,
     ) -> Result<PayjoinProposal, Error> {
+        let sender_input_indexes = self.get_sender_input_indexes();
+        for i in sender_input_indexes {
+            log::trace!("Clearing sender script witness for input {}", i);
+            self.payjoin_psbt.inputs[i].final_script_witness = None;
```

Comment:
should final_script_sig and tap_key_sig also be removed at this step? Might they cause problems if they were not removed before wallet_process_psbt of various implementations?

Reply:
It seems reasonable to remove all signatures at this step as they would be invalid anyways. Updated and smoke-tested with another test receive.